nimble/controller/src/ble_ll_ctrl.c

```diff
@@ -689,8 +689,8 @@ ble_ll_ctrl_rx_phy_req(struct ble_ll_conn_sm *connsm, uint8_t *req,
         /* XXX: TODO: if we started another procedure with an instant
          * why are we doing this? Need to look into this.*/
 
-        /* Respond to master's phy update procedure */
-        CONN_F_PEER_PHY_UPDATE(connsm) = 1;
+        /* Let us wait for LL_PHY_UPDATE_IND */
+        ble_ll_ctrl_proc_start(connsm, BLE_LL_CTRL_PROC_PHY_UPDATE);
```

Comment:
seems like this only usually works because current procedure is non-idle when this is called - only bitmask is set in such case which is what you want. in case if above is executed current procedure will be set to idle and this will start an actualy phy update procedure.

Reply:
hmm you right.. will remove this patch now. need to figure out how to do it nicely.